utils/search/backends/solr555pysolr.py

```diff
@@ -358,10 +149,302 @@ def get_forum_index(self):
                 always_commit=True
             )
         return self.forum_index
+    
+    # Util functions
+    def transform_document_into_update_document(self, document):
```

Comment:
It is true that we don't use `self` but it made sense to have it as a method because in this way it is easy to override methods in subclasses. This method does actually not need to be overwritten, but there are other similar methods that don't use `self` either but that need to be overwritten (e.g. `search_process_filter`). I decided to keep them all as methods for consistency.